extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/v1/GreenbidsRealTimeDataProcessedAuctionRequestHook.java

```diff
@@ -0,0 +1,210 @@
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.analytics.TagsImpl;
+import org.prebid.server.hooks.v1.InvocationAction;
+import org.prebid.server.hooks.v1.InvocationResult;
+import org.prebid.server.hooks.v1.InvocationStatus;
+import org.prebid.server.hooks.v1.analytics.Result;
+import org.prebid.server.hooks.v1.analytics.Tags;
+import org.prebid.server.hooks.v1.auction.AuctionInvocationContext;
+import org.prebid.server.hooks.v1.auction.AuctionRequestPayload;
+import org.prebid.server.hooks.v1.auction.ProcessedAuctionRequestHook;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class GreenbidsRealTimeDataProcessedAuctionRequestHook implements ProcessedAuctionRequestHook {
+
+    private static final String CODE = "greenbids-real-time-data-processed-auction-request";
+    private static final String ACTIVITY = "greenbids-filter";
+    private static final String SUCCESS_STATUS = "success";
+    private static final String BID_REQUEST_ANALYTICS_EXTENSION_NAME = "greenbids-rtd";
+
+    private final ObjectMapper mapper;
+    private final FilterService filterService;
+    private final OnnxModelRunnerWithThresholds onnxModelRunnerWithThresholds;
+    private final GreenbidsInferenceDataService greenbidsInferenceDataService;
+    private final GreenbidsInvocationService greenbidsInvocationService;
+
+    public GreenbidsRealTimeDataProcessedAuctionRequestHook(
+            ObjectMapper mapper,
+            FilterService filterService,
+            OnnxModelRunnerWithThresholds onnxModelRunnerWithThresholds,
+            GreenbidsInferenceDataService greenbidsInferenceDataService,
+            GreenbidsInvocationService greenbidsInvocationService) {
+        this.mapper = Objects.requireNonNull(mapper);
+        this.filterService = Objects.requireNonNull(filterService);
+        this.onnxModelRunnerWithThresholds = Objects.requireNonNull(onnxModelRunnerWithThresholds);
+        this.greenbidsInferenceDataService = Objects.requireNonNull(greenbidsInferenceDataService);
+        this.greenbidsInvocationService = Objects.requireNonNull(greenbidsInvocationService);
+    }
+
+    @Override
+    public Future<InvocationResult<AuctionRequestPayload>> call(
+            AuctionRequestPayload auctionRequestPayload,
+            AuctionInvocationContext invocationContext) {
+
+        final AuctionContext auctionContext = invocationContext.auctionContext();
+        final BidRequest bidRequest = auctionContext.getBidRequest();
+        final Partner partner = parseBidRequestExt(bidRequest);
+
+        if (partner == null) {
+            return Future.succeededFuture(toInvocationResult(
+                    bidRequest, null, InvocationAction.no_action));
+        }
+
+        return Future.join(
```

Comment:
Do we really need a `join` operation here? Maybe `all` would be better?

join - will wait for all responses
all - will fail on first fail

Reply:
Here I've used the Future.join following @AntoxaAntoxic [comment](https://github.com/prebid/prebid-server-java/pull/3242#discussion_r1743301448)

Reply:
@EvgeniiMunin use `all` instead